### PR TITLE
Feature/hooks refactor

### PR DESCRIPTION
**Pull request by MrBrN197**
## Changes | 🚀

- ☑ Refactor `Calculator` component to functional component
- ☑ Refactor `App` component to functional component

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
 import Calculator from './components/Calculator';
 import calculate from './logic/calculate';
 
-class App extends react.Component {
-  constructor(props) {
-    super(props);
-    this.state = { total: null, next: null, operation: null };
-    this.updateDetails = this.updateDetails.bind(this);
-  }
+function App() {
```
Comment (rukundoeric) [inline src/App.js]: - [x] <b>REQUIRED:<b/> Prefer using ES6 arrow function. Actually, there is nothing wrong with ES5 functions syntax but it is mandatory to use es6 features in the javascript module. besides, Arrow functions provide a simplified structure, allowing for a more concise way to write function expressions. which makes them better than ES5 functions(function declarations). Here is the syntax of an ES6 arrow function...
```javascript
  const functionName = () => {
     // code here
  }
```

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,44 +1,35 @@
...
-          <button onClick={updateDetails} className="orange" type="button">&minus;</button>
-          <button onClick={updateDetails} type="button">1</button>
-          <button onClick={updateDetails} type="button">2</button>
-          <button onClick={updateDetails} type="button">3</button>
-          <button onClick={updateDetails} className="orange" type="button">+</button>
-          <button onClick={updateDetails} className="double" type="button">0</button>
-          <button onClick={updateDetails} type="button">.</button>
-          <button onClick={updateDetails} className="orange" type="button">=</button>
-        </div>
+function Calculator({ updateDetails, value }) {
```
Comment (rukundoeric) [inline src/components/Calculator.js]: - [x] <b>REQUIRED:<b/> Prefer using ES6 arrow function. Actually, there is nothing wrong with ES5 functions syntax but it is mandatory to use es6 features in the javascript module. besides, Arrow functions provide a simplified structure, allowing for a more concise way to write function expressions. which makes them better than ES5 functions(function declarations). Here is the syntax of an ES6 arrow function...
```javascript
  const functionName = () => {
     // code here
  }
```

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @MrBrN197, 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 

## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - [ ] Check the comments under the review.

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/Calculator.js (hunk 2 above)
Comment (MrBrN197) [reply src/components/Calculator.js]: You are right I agree.

**Review by dasileker — APPROVED** [review]
## **Hi @MrBrN197,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**